### PR TITLE
Ability to pop to a screen in the stack

### PR DESCRIPTION
**Pull request by mattyclarkson**
```ts
import as React from 'react';
import { pop } from '@ef-carbon/react-native-navigation';
export class MyScreen extends React.PureComponent {
  private readonly navigateToNavigation = (): void => {
    // Pop all screen that are above this screen in the stack
    pop(MyScreen);
    // Pop last screen
    pop();
    // Pop multiple screens
    pop(2);
  }
}
```

### CODE REVIEW

File: lib/pop.ts [hunk 1]
```diff
@@ -1,27 +1,58 @@
...
+}
 
 /**
- * Pushes a new screen onto the stack.
- * @param screen the screen to push onto the stack
- * @param props the type checked properties for the screen
- * @returns the actual route that was navigated to
+ * Pops a screen, or multiple, from the stack
+ * @param count the number of screens to pop from the stack. This can also be a screen that is already in the stack;
+ * all screens up to that screen will be popped
```
Comment (ionTea) [inline lib/pop.ts]: maybe document that it throws if screen is missing on the stack

File: lib/pop.ts [hunk 2]
```diff
@@ -1,27 +1,58 @@
...
+ *     pop(2);
  *   }
  * }
  * ```
  */
-export function pop(count: number = 1): void {
+export function pop(count: number | ScreenComponentClass = 1): void {
+  if (typeof count !== 'number') {
+    const { length } = stack;
+    const index = findLastIndex(stack, ({ screen }) => count === screen);
```
Comment (ionTea) [inline lib/pop.ts]: I stared at this way to long believing 'count' was a number, and you compared screen to a number.

Comment (mattyclarkson) [reply lib/pop.ts]: I'll update the variable to `countOrScreen`

File: lib/pop.ts (hunk 1 above)
Comment (mattyclarkson) [reply lib/pop.ts]: Good shout, will update.

**Pull request merged**